### PR TITLE
Document using CLI to provision clusters using Gardener

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Document using CLI to provision clusters using Gardener

**Related issue(s)**
See also #6353

### CODE REVIEW

File: docs/kyma/04-04-cluster-installation.md [hunk 1]
```diff
@@ -156,25 +156,39 @@ This installation guide explains how you can quickly deploy Kyma on a cluster wi
...
+   ```
+
+   To provision an AKS cluster, run:
+
+   ```
+   kyma provision gardener --target-provider azure -n {cluster_name} -p {project_name} -s {kyma_gardener_azure_secret} -c {path_to_gardener_kubeconfig} -t Standard_D2_v3 --region westeurope --disk-size 35 --disk-type Standard_LRS --extra vnetcidr="10.250.0.0/19"
+   ```
+   For a complete list of flags and their descriptions, see [this](https://github.com/kyma-project/cli/blob/master/docs/gen-docs/kyma_provision_gardener.md) document.
+
+3. After you provision the cluster, its `kubeconfig` file will be downloaded automatically for you to use.
```
Comment (clebs) [inline docs/kyma/04-04-cluster-installation.md]:
```suggestion
3. After you provision the cluster, its `kubeconfig` file will be downloaded and set as current context automatically for you to use.
```

File: docs/kyma/04-04-cluster-installation.md [hunk 2]
```diff
@@ -156,25 +156,39 @@ This installation guide explains how you can quickly deploy Kyma on a cluster wi
...
 
-   | Tab  |  Setting |  Required value |
-   |---|---|---|
-   | Cluster Details |  Kubernetes Version | `1.15.6`  |
-   | Worker  |  Machine type | `n1-standard-4` (GCP) `Standard_D4_v3` (Azure)|
-   | Worker  | Autoscaler min.  | `3` |
+    For AKS:
+    * Create a project in Gardener.
+    * Add the Azure Secret under **Secrets** in the Gardener dashboard. Use the details of your Azure service account. If do not have an account, you need to request one.
+    * Add the service account and download the Gardener `kubeconfig` file.
```
Comment (kazydek) [inline docs/kyma/04-04-cluster-installation.md]:
```suggestion
    * Add the service account and download the Gardener's `kubeconfig` file.
```

File: docs/kyma/04-04-cluster-installation.md [hunk 3]
```diff
@@ -156,25 +156,39 @@ This installation guide explains how you can quickly deploy Kyma on a cluster wi
...
 
-2. Provision a cluster form the **Clusters** tab. Click the plus sign in the lower-right corner and choose the infrastructure in which you want to provision your cluster. Apply these settings in the following tabs:
+    >**NOTE**: You need to perform these steps only once.
+   
+    For GCP:
+    * Create a project in Gardener.
+    * Add a [new service account and roles](https://gardener.cloud/050-tutorials/content/howto/gardener_gcp/#create-a-new-serviceaccount-and-assign-roles).
+      >**NOTE**: Run gcloud auth application-default login and log in with your Google Cloud credentials to get authenticated by Google SDK.
+    * Add the GCP Secret under **Secrets** in the Gardener dashboard.
+    * Add the service account and download the Gardener `kubeconfig` file.
```
Comment (kazydek) [inline docs/kyma/04-04-cluster-installation.md]:
```suggestion
    * Add the service account and download the Gardener's `kubeconfig` file.
```

File: docs/kyma/04-04-cluster-installation.md [hunk 4]
```diff
@@ -156,25 +156,39 @@ This installation guide explains how you can quickly deploy Kyma on a cluster wi
...
-1. In the left navigation of the Gardener UI, go to the **Secrets** tab and add Secrets to enable provisioning clusters on different architectures. To learn about the requirements for each environment, click the question mark buttons.
+1. Use the Gardener dashboard to configure provider settings.
 
-2. Provision a cluster form the **Clusters** tab. Click the plus sign in the lower-right corner and choose the infrastructure in which you want to provision your cluster. Apply these settings in the following tabs:
+    >**NOTE:** You need to perform these steps only once.
+   
+    * For GCP:
+      * Create a project in Gardener.
+      * Add a [new service account and roles](https://gardener.cloud/050-tutorials/content/howto/gardener_gcp/#create-a-new-serviceaccount-and-assign-roles).
+      >**NOTE**: Run `gcloud auth application-default login` and log in with your Google Cloud credentials to be authenticated by Google SDK.
```
Comment (a-thaler) [inline docs/kyma/04-04-cluster-installation.md]: That note is not relevant here as we will not use google specific tooling

File: docs/kyma/04-04-cluster-installation.md [hunk 5]
```diff
@@ -39,7 +39,7 @@ This installation guide explains how you can quickly deploy Kyma on a cluster wi
   Gardener
   </summary>
 
-- [Gardener](https://gardener.cloud/) seed cluster
+- [Gardener](https://gardener.cloud/) account
 - [Google Cloud Platform](https://console.cloud.google.com/) (GCP) project with Kubernetes Engine API enabled or a [Microsoft Azure](https://azure.microsoft.com) account
```
Comment (a-thaler) [inline docs/kyma/04-04-cluster-installation.md]: For Gardener on GCP we need a GCP project, probably having the Compute Engine API enabled but we can skip that, definitly not the kubernetes Engine API.
For Gardener on Azure we need an Azure project

File: docs/kyma/04-04-cluster-installation.md [hunk 6]
```diff
@@ -156,25 +156,39 @@ This installation guide explains how you can quickly deploy Kyma on a cluster wi
...
-   ```bash
-   export KUBECONFIG={PATH_TO_KUBECONFIG_FILE}
    ```
+   kyma provision gardener -n {cluster_name} -p {project_name} -s {kyma_gardener_gcp_secret} -c {path_to_gardener_kubeconfig}
+   ```
+
+   To provision an AKS cluster, run:
+
+   ```
+   kyma provision gardener --target-provider azure -n {cluster_name} -p {project_name} -s {kyma_gardener_azure_secret} -c {path_to_gardener_kubeconfig} -t Standard_D2_v3 --region westeurope --disk-size 35 --disk-type Standard_LRS --extra vnetcidr="10.250.0.0/19"
```
Comment (a-thaler) [inline docs/kyma/04-04-cluster-installation.md]: Maybe name kyma_gardener_azure_secret as kyma_gardener_azure_secret_name to indicate that it is not the actual secret you are passing but the name of the secret which you have configured before
Same for GCP call

File: docs/kyma/04-04-cluster-installation.md (hunk 5 above)
Comment (bszwarc) [reply docs/kyma/04-04-cluster-installation.md]: Changed :)

File: docs/kyma/04-04-cluster-installation.md (hunk 4 above)
Comment (bszwarc) [reply docs/kyma/04-04-cluster-installation.md]: Removed :)

File: docs/kyma/04-04-cluster-installation.md (hunk 6 above)
Comment (bszwarc) [reply docs/kyma/04-04-cluster-installation.md]: Changed :)

**Pull request merged**